VirtoCommerce.CustomerModule.Data/Handlers/LogChangesMemberChangedEventHandler.cs

```diff
@@ -0,0 +1,249 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using AutoCompare;
+using VirtoCommerce.Domain.Commerce.Model;
+using VirtoCommerce.Domain.Common.Events;
+using VirtoCommerce.Domain.Customer.Events;
+using VirtoCommerce.Domain.Customer.Model;
+using VirtoCommerce.Domain.Customer.Services;
+using VirtoCommerce.Domain.Order.Model;
+using VirtoCommerce.Platform.Core.ChangeLog;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.Events;
+
+namespace VirtoCommerce.CustomerModule.Data.Handlers
+{
+    public class LogChangesMemberChangedEventHandler : IEventHandler<MemberChangedEvent>
+    {
+        private readonly IChangeLogService _changeLogService;
+        private static readonly string[] ObservedProperties;
+
+        static LogChangesMemberChangedEventHandler()
+        {
+            var memberPropsNames = ReflectionUtility.GetPropertyNames<Contact>(
+                x => x.Name, x => x.FirstName, x => x.LastName, x => x.MiddleName,
+                x => x.Salutation, x => x.FullName, x => x.BirthDate);
+            ObservedProperties = memberPropsNames.ToArray();
+        }
+
+        public LogChangesMemberChangedEventHandler(IChangeLogService changeLogService)
+        {
+            _changeLogService = changeLogService;
+        }
+
+        public virtual Task Handle(MemberChangedEvent message)
+        {
+            var operationLogs =
```

Comment:
We need handle only Contact members, so I think we can add cast and filtration on this level.

Reply:
Filter - yes, but we need not just cast. `message.ChangedEntries` is of type `IEnumeratble<GenericChangedEntry<Member>>`, so to cast we would need to create new `GenericChangedEntry<Contact>`. I don't think it's necessary to create new objects, since I unwrap it on line 55:
```csharp
if (changedEntry.OldEntry is Contact contact)
```